internal/frr/config.go

```diff
@@ -187,11 +187,28 @@ func templateConfig(data interface{}) (string, error) {
 			"deniedIncomingList": func(neighbor *NeighborConfig) string {
 				return fmt.Sprintf("%s-denied-inpl-%s", neighbor.ID(), neighbor.IPFamily)
 			},
-			"mustDisableConnectedCheck": func(ipFamily ipfamily.Family, myASN, asn uint32, eBGPMultiHop bool) bool {
-				// return true only for IPv6 eBGP sessions
-				if ipFamily == "ipv6" && myASN != asn && !eBGPMultiHop {
+			"mustDisableConnectedCheck": func(ipFamily ipfamily.Family, myASN uint32, asn string, eBGPMultiHop bool) bool {
+				// return true only for non-multihop IPv6 eBGP sessions
+				if ipFamily == ipfamily.IPv4 {
+					return false
+				}
+
+				if eBGPMultiHop {
+					return false
+				}
+
+				if asn == "internal" {
```

Comment:
let's either have a comment saying that all these ifs mean it's external (or wrap in a "isEBGP" helper)

Reply:
done